### PR TITLE
Generic auth(IAM)

### PR DESCRIPTION
**Pull request by tejash-jl**
https://github.com/Sunbird-RC/community/issues/373

### CODE REVIEW

File: java/middleware/registry-middleware/auth0/pom.xml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    <dependencyManagement>
+        <dependencies>
+
+        </dependencies>
+    </dependencyManagement>
+    <dependencies>
+        <dependency>
+            <groupId>com.auth0</groupId>
+            <artifactId>auth0</artifactId>
```
Comment (prasanna-egov) [inline java/middleware/registry-middleware/auth0/pom.xml:22]: Is this specific to Auth0? Or did we just use the library?

Comment (tejash-jl) [reply java/middleware/registry-middleware/auth0/pom.xml:22]: this is a auth0 library used for creating users in auth0 platform,
one should implement the logic based on the interfaces provided to support creating users in different IAM portals

**Review by srprasanna — COMMENTED** [review]
@tejash-jl  we will also need to add documentation on adding OAuthProviders and Identitymanagers from the interfaces. Some documentation will help.

File: java/middleware/registry-middleware/authorization/src/main/java/dev/sunbirdrc/registry/authorization/CustomJwtAuthenticationConverter.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.jwt.Jwt;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+class CustomJwtAuthenticationConverter implements Converter<Jwt, AbstractAuthenticationToken> {
```
[the remark this answers is not in the document]
Comment (tejash-jl) [reply java/middleware/registry-middleware/authorization/src/main/java/dev/sunbirdrc/registry/authorization/CustomJwtAuthenticationConverter.java:17]: yes, we can use that, 
going forward we can leverage it

File: Makefile [hunk 3]
```diff
@@ -74,6 +74,19 @@ test: build
 	@cd java/apitest && MODE=events ../mvnw -Pe2e test || echo 'Tests failed'
 	@docker-compose down
 	@rm -rf db-data-5 || echo "no permission to delete"
+	# test with fusionauth
+	@RELEASE_VERSION=latest DB_DIR=db-data-6 SEARCH_PROVIDER_NAME=dev.sunbirdrc.registry.service.NativeSearchService FUSION_WRAPPER_BUILD=services/sample-fusionauth-service/ FUSIONAUTH_ISSUER_URL=http://fusionauth:9011/ oauth2_resource_uri=http://fusionauth:9011/ oauth2_resource_roles_path=roles identity_provider=dev.sunbirdrc.auth.genericiam.AuthProviderImpl sunbird_sso_url=http://fusionauthwrapper:3990/fusionauth/api/v1/user IMPORTS_DIR=services/sample-fusionauth-service/imports docker-compose -f docker-compose.yml -f services/sample-fusionauth-service/docker-compose.yml up -d db es fusionauth fusionauthwrapper
```
[the remark this answers is not in the document]
Comment (tejash-jl) [reply Makefile:78]: this is for e2e test with fusionauth

File: java/middleware/registry-middleware/authorization/src/main/java/dev/sunbirdrc/registry/authorization/pojos/OAuth2Properties.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package dev.sunbirdrc.registry.authorization.pojos;
+
+
+import lombok.Data;
+
+@Data
+public class OAuth2Properties {
+	private String rolesPath = "realm_access.roles";
```
[the remark this answers is not in the document]
Comment (tejash-jl) [reply java/middleware/registry-middleware/authorization/src/main/java/dev/sunbirdrc/registry/authorization/pojos/OAuth2Properties.java:8]: yes, this is from config/env itself,
these are the default values

**Review by srprasanna — APPROVED** [review]

**Pull request merged**